### PR TITLE
[Pub/sub] Add match example to API spec

### LINKED ISSUE #2923: [PubSub] Missing route event via 'match' example (programmatic subscriptions)
I believe this payload is not complete https://docs.dapr.io/reference/api/pubsub_api/#http-response-body

```json
[
  {
    "pubsubname": "pubsub",
    "topic": "newOrder",
    "route": "/orders",
    "metadata": {
      "rawPayload": "true",
    }
  }
]
```

I believe it is missing some additional properties for doing expression matches. See the JavaScript example [here](https://docs.dapr.io/developing-applications/building-blocks/pubsub/subscription-methods/#programmatic-subscriptions)

image

### PR DESCRIPTION
**Pull request by hhunter-ms**
## Description

Add matches example to the pub/sub api spec

## Issue reference

PR will close: #2923

### CODE REVIEW

**Review by yaron2 — DISMISSED** [review]

File: daprdocs/content/en/reference/api/pubsub_api.md [hunk 1]
```diff
@@ -177,7 +177,14 @@ Example:
...
     "pubsubname": "pubsub",
     "topic": "newOrder",
-    "route": "/orders",
+    "route": {
+      "rules": [
+        {
+          "match": "event.type == order",
+          "path": "/orders"
+        }
+      ]
```
Comment (msfussell) [inline daprdocs/content/en/reference/api/pubsub_api.md:186]:
```suggestion
      ]
      default: /otherorders
```

File: daprdocs/content/en/reference/api/pubsub_api.md [hunk 2]
```diff
@@ -177,7 +177,14 @@ Example:
   {
     "pubsubname": "pubsub",
     "topic": "newOrder",
-    "route": "/orders",
+    "route": {
```
Comment (msfussell) [inline daprdocs/content/en/reference/api/pubsub_api.md]:
```suggestion
    "routes": {
```

**Comment by msfussell** [comment]
@hhunter-ms There need to be several more changes here. 
Mostly the update here should follow this article https://docs.dapr.io/developing-applications/building-blocks/pubsub/howto-route-messages/
1) See comments above on `routes` not `route`
2) In the https://docs.dapr.io/reference/api/pubsub_api/#provide-routes-for-dapr-to-deliver-topic-events section after the "In order to deliver topic events, a POST call will be made to user code with the route specified in the subscription response." sentence, say that under `routes` you can provide `rules` that `match` a certain condition to specific path when a message topic is received . I would also mention that you can provide a `default` route for any rules that do not have a specific match.
3) I would then add a link to https://docs.dapr.io/developing-applications/building-blocks/pubsub/howto-route-messages
4) Whilst we are fixing this up, can you also update
-  https://docs.dapr.io/developing-applications/building-blocks/pubsub/howto-route-messages#programmatic-subscription here from saying "Node" in the tab to "Javascript" and the C# one to .NET, so this is consistent with https://docs.dapr.io/developing-applications/building-blocks/pubsub/subscription-methods/#programmatic-subscriptions
- Generally what is the name and order we have been using for languages?

File: daprdocs/content/en/reference/api/pubsub_api.md [hunk 3]
```diff
@@ -177,7 +177,15 @@ Example:
...
     "topic": "newOrder",
-    "route": "/orders",
+    "routes": {
+      "rules": [
+        {
+          "match": "event.type == order",
+          "path": "/orders"
+        }
+      ]
+      default: /otherorders
```
Comment (msfussell) [inline daprdocs/content/en/reference/api/pubsub_api.md]:
```suggestion
      "default" : "/otherorders"
```

**Review by msfussell — APPROVED** [review]
Did a fix up on the default value

**Pull request merged**